doctor/views.py

```diff
@@ -39,25 +39,42 @@ def create(self, request, *args, **kwargs):
     def get_data_with_userid(self, request):
         return {'user_id': request.user.id, **request.data}
 
-class DoctorAPI(ReadOnlyModelViewSet):
+class ReadOnlyDoctorAPI(ReadOnlyModelViewSet):
     """ 의사 조회 API
-        의사 정보를 조회하는 API
+        비로그인 상태 의사 정보를 조회하는 API
         Note:
-            * Authorization이 존재하는 경우
-                * Authorization에 해당하는 의사의 정보를 조회한다.
-            * Authorization이 존재하지 않는 경우
-                * 모든 의사의 정보를 조회한다.
+            * List, Retrieve
+                List        모든 의사의 정보를 조회한다.
+                Retreive    특정 의사의 정보를 조회한다. Path params 사용.
             * GET method만 허용한다.
     """
-    permission_classes = [IsAuthenticatedOrReadOnly]
-    serializer_class = DoctorSerializer
+    permission_classes = [AllowAny]
+    serializer_class = ReadOnlyDoctorSerializer
 
     def get_queryset(self):
-        try:
-            return Doctor.objects.filter(user=self.request.user)
-        except:
             return Doctor.objects.all()
 
+class DoctorInfoAPI(mixins.RetrieveModelMixin,
+                    GenericAPIView):
+    """ 의사 조회 API
+        로그인 상태 의사 정보를 조회하는 API
+        Note:
+            * Retrieve 현재 로그인한 본인(의사)의 정보를 조회한다.
+            * GET method만 허용한다.
+    """
+    queryset = Doctor.objects.all()
+    permission_classes = [IsAuthenticated]
+    serializer_class = DoctorSerializer
+    
+    def get(self, request, *args, **kwargs):
+        return self.retrieve(request, *args, **kwargs)
+    
+    def get_object(self, queryset=None):
+        if queryset is None:
+            queryset = self.get_queryset()
+        
+        return queryset.filter(user=self.request.user).get()
```

Comment:
```suggestion
        return self.request.user.doctor
```
permission으로 의사 구분할거면 이렇게 해도 될 것 같고 아니라면
get_object_or_404라고 원래 get_object에서 사용하는 없으면 404 not found 바로 리턴하는 거 있는데 그거 써도 좋을 것 같아요.